tests/models/test_model_input_examples.py

```diff
@@ -55,40 +65,53 @@ def test_input_examples(pandas_df_with_all_types):
             .all()
         )
 
+    # NB: Choose nparrays columns that can be encoded using proto_json_utils.pyNumpyEncoder
+    new_df = pandas_df_with_all_types[["integer", "long", "float", "double"]]
+
     # pass the input as dictionary instead
     with TempDir() as tmp:
-        d = {
-            name: pandas_df_with_all_types[name].values for name in pandas_df_with_all_types.columns
-        }
+        d = {name: new_df[name].values for name in new_df.columns}
         example = _Example(d)
         example.save(tmp.path())
         filename = example.info["artifact_path"]
-        parsed_df = _dataframe_from_json(tmp.path(filename), sig.inputs)
-        assert (pandas_df_with_all_types == parsed_df).all().all()
-
-    # input passed as numpy array
-    sig = infer_signature(pandas_df_with_all_types.values)
-    with TempDir() as tmp:
-        example = _Example(pandas_df_with_all_types.values)
-        example.save(tmp.path())
-        filename = example.info["artifact_path"]
         with open(tmp.path(filename), "r") as f:
             data = json.load(f)
-            assert set(data.keys()) == set(("data",))
-        parsed_ary = _dataframe_from_json(tmp.path(filename), schema=sig.inputs).values
-        assert (pandas_df_with_all_types.values == parsed_ary).all().all()
+            assert set(data.keys()) == set(("inputs",))
+        parsed_dict = _read_tensor_input_from_json(tmp.path(filename))
+        assert d.keys() == parsed_dict.keys()
+        for key in d:
+            assert np.array_equal(d[key], parsed_dict[key])
 
-    # pass multidimensional array
-    with TempDir() as tmp:
-        example = np.array([[[1, 2, 3]]])
-        with pytest.raises(TensorsNotSupportedException):
-            _Example(example)
+    # input passed as numpy array
+    for col in new_df:
+        input_example = new_df[col].to_numpy()
+        with TempDir() as tmp:
+            example = _Example(input_example)
+            example.save(tmp.path())
+            filename = example.info["artifact_path"]
+            with open(tmp.path(filename), "r") as f:
+                data = json.load(f)
+                assert set(data.keys()) == set(("instances",))
+            parsed_ary = _read_tensor_input_from_json(tmp.path(filename))
+            assert np.array_equal(parsed_ary, input_example)
 
     # pass multidimensional array
-    with TempDir() as tmp:
-        example = np.array([[1, 2, 3]])
-        with pytest.raises(TensorsNotSupportedException):
-            _Example({"x": example, "y": example})
+    for col in dict_of_ndarrays:
+        input_example = dict_of_ndarrays[col]
+        with TempDir() as tmp:
+            example = _Example(input_example)
+            example.save(tmp.path())
+            filename = example.info["artifact_path"]
+            with open(tmp.path(filename), "r") as f:
+                data = json.load(f)
+                assert set(data.keys()) == set(("instances",))
```

Comment:
nit: I think we should drop this assert. the important part is it parses back in correctly